src/components/syncplay/syncplayManager.js

```diff
@@ -0,0 +1,737 @@
+     */
+    disableSyncplay (showMessage = false) {
+        this.syncplayEnabledAt = null;
+        this.syncplayReady = false;
+        this.lastCommand = null;
+        this.queuedCommand = null;
+        this.syncEnabled = false;
+        events.trigger(this, "SyncplayEnabled", [false]);
+        this.restorePlaybackManager();
+
+        if (showMessage) {
+            toast({
+                text: globalize.translate('MessageSyncplayDisabled')
+            });
+        }
+    }
+
+    /**
+     * Gets Syncplay status.
+     * @returns {boolean} _true_ if user joined a group, _false_ otherwise.
+     */
+    isSyncplayEnabled () {
+        return this.syncplayEnabledAt !== null ? true : false;
+    }
+
+    /**
+     * Schedules a resume playback on the player at the specified clock time.
+     * @param {Date} playAtTime The server's UTC time at which to resume playback.
+     * @param {number} positionTicks The PositionTicks from where to resume.
+     */
+    schedulePlay (playAtTime, positionTicks) {
+        this.clearScheduledCommand();
+        var currentTime = new Date();
+        var playAtTimeLocal = timeSyncManager.serverDateToLocal(playAtTime);
+
+        if (playAtTimeLocal > currentTime) {
+            var playTimeout = playAtTimeLocal - currentTime;
+            playbackManager.syncplay_seek(positionTicks);
+
+            this.scheduledCommand = setTimeout(() => {
+                playbackManager.syncplay_unpause();
+
+                this.syncTimeout = setTimeout(() => {
+                    this.syncEnabled = true;
+                }, SyncMethodThreshold / 2);
+
+            }, playTimeout);
+
+            // console.debug("Syncplay schedulePlay:", playTimeout);
+        } else {
+            // Group playback already started
+            var serverPositionTicks = positionTicks + (currentTime - playAtTimeLocal) * 10000;
+            playbackManager.syncplay_unpause();
+            playbackManager.syncplay_seek(serverPositionTicks);
+
+            this.syncTimeout = setTimeout(() => {
+                this.syncEnabled = true;
+            }, SyncMethodThreshold / 2);
+        }
+    }
+
+    /**
+     * Schedules a pause playback on the player at the specified clock time.
+     * @param {Date} pauseAtTime The server's UTC time at which to pause playback.
+     * @param {number} positionTicks The PositionTicks where player will be paused.
+     */
+    schedulePause (pauseAtTime, positionTicks) {
+        this.clearScheduledCommand();
+        var currentTime = new Date();
+        var pauseAtTimeLocal = timeSyncManager.serverDateToLocal(pauseAtTime);
+
+        if (pauseAtTimeLocal > currentTime) {
+            var pauseTimeout = pauseAtTimeLocal - currentTime;
+
+            this.scheduledCommand = setTimeout(() => {
+                playbackManager.syncplay_pause();
+                setTimeout(() => {
+                    playbackManager.syncplay_seek(positionTicks);
+                }, 800);
```

Comment:
Hacky way to give the player some time to pause before seeking, otherwise it just breaks. I should probably try to see if replacing it with `waitForEventOnce` works in a reliable way.